src/node/internal/internal_fs_promises.ts

```diff
@@ -149,72 +146,80 @@ export class FileHandle extends EventEmitter {
-    offsetOrOptions: number | ReadSyncOptions = {},
+  read<T extends NodeJS.ArrayBufferView>(
+    bufferOrOptions: T | ReadAsyncOptions<T> = {},
+    offsetOrOptions: number | ReadAsyncOptions<T> = {},
     length?: number,
     position: Position = null
   ): Promise<{ bytesRead: number; buffer: T }> {
-    if (this.#fd === undefined) {
-      throw new ERR_EBADF({ syscall: 'stat' });
-    }
+    try {
+      if (this.#fd === undefined) {
+        throw new ERR_EBADF({ syscall: 'stat' });
+      }
 
-    let options: any;
-    if (isArrayBufferView(bufferOrOptions)) {
-      if (typeof offsetOrOptions === 'number') {
-        options = {
-          buffer: bufferOrOptions,
-          offset: offsetOrOptions,
-          length: length ?? bufferOrOptions.byteLength,
-          position: position ?? null,
-        };
+      let options: ReadAsyncOptions<T>;
+      if (isArrayBufferView(bufferOrOptions)) {
+        if (typeof offsetOrOptions === 'number') {
+          options = {
+            buffer: bufferOrOptions,
+            offset: offsetOrOptions,
+            length: length ?? bufferOrOptions.byteLength,
+            position: position ?? null,
+          };
+        } else {
+          options = {
+            buffer: bufferOrOptions,
+            ...offsetOrOptions,
+          };
+        }
       } else {
-        options = {
-          buffer: bufferOrOptions,
-          ...offsetOrOptions,
-        };
+        options = bufferOrOptions;
       }
-    } else {
-      options = bufferOrOptions;
-    }
 
-    const {
-      buffer = Buffer.alloc(16384),
-      offset: actualOffset = buffer.byteOffset,
-      length: actualLength = buffer.byteLength - buffer.byteOffset,
-      position: actualPosition = null,
-    } = options || {};
-
-    validateUint32(actualOffset, 'offset');
-    validateUint32(actualLength, 'length');
-    validatePosition(actualPosition, 'position');
-
-    const bytesRead = fssync.readSync(
-      this.#fd,
-      buffer,
-      actualOffset,
-      actualLength,
-      actualPosition
-    );
-
-    return {
-      bytesRead,
-      buffer,
-    };
+      const {
+        buffer = Buffer.alloc(16384),
+        offset: actualOffset = buffer.byteOffset,
+        length: actualLength = buffer.byteLength - buffer.byteOffset,
+        position: actualPosition = null,
+      } = options;
+
+      validateUint32(actualOffset, 'offset');
+      validateUint32(actualLength, 'length');
+      validatePosition(actualPosition, 'position');
+
+      const bytesRead = fssync.readSync(
+        this.#fd,
+        buffer as NodeJS.ArrayBufferView,
+        actualOffset,
+        actualLength,
+        actualPosition
+      );
+
+      return Promise.resolve({
+        bytesRead,
+        buffer: buffer as T,
+      });
+    } catch (err) {
+      return Promise.reject(err as Error);
+    }
   }
 
-  async readv<T extends NodeJS.ArrayBufferView>(
+  readv<T extends NodeJS.ArrayBufferView>(
```

Comment:
Why do we remove the async? It behaves the same way without the try/catch? Isn't this better?

Reply:
to keep the typescript linter from complaining about an async function with no await.